lib0/src/any.rs

```diff
@@ -13,3 +13,131 @@ pub enum Any {
     Array(Vec<Any>),
     Map(HashMap<String, Any>),
 }
+
+impl std::fmt::Display for Any {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Any::Null => write!(f, "null")?,
+            Any::Undefined => write!(f, "undefined")?,
+            Any::Bool(value) => write!(f, "{}", value)?,
+            Any::Number(value) => write!(f, "{}", value)?,
+            Any::BigInt(value) => write!(f, "{}", value)?,
+            Any::String(value) => write!(f, "\"{}\"", value)?,
+            Any::Buffer(value) => write!(f, "[binary: {} bytes]", value.len())?, //TODO: use base64?
+            Any::Array(values) => {
+                write!(f, "[")?;
+                let mut i = values.iter();
+                if let Some(value) = i.next() {
+                    value.fmt(f)?;
+                }
+                while let Some(value) = i.next() {
+                    write!(f, ", ")?;
+                    value.fmt(f)?;
+                }
+                write!(f, "]")?;
+            },
+            Any::Map(entries) => {
+                write!(f, "{{")?;
+                let mut i = entries.iter();
+                if let Some((key,value)) = i.next() {
+                    write!(f, "\"{}\": {}", key, value)?;
+                }
+                while let Some((key,value)) = i.next() {
+                    write!(f, ", \"{}\": {}", key, value)?;
+                }
+                write!(f, "}}")?;
+            },
+        }
+
+        Ok(())
+    }
+}
+
+impl Into<Any> for bool {
+    fn into(self) -> Any {
+        Any::Bool(self)
+    }
+}
+
+impl Into<Any> for f64 {
+    fn into(self) -> Any {
+        Any::Number(self)
+    }
+}
+
+impl Into<Any> for f32 {
+    fn into(self) -> Any {
+        Any::Number(self as f64)
+    }
+}
+
+impl Into<Any> for u32 {
+    fn into(self) -> Any {
+        Any::BigInt(self as i64)
```

Comment:
This should convert to `Any::Number(self as f64)` instead because `Any:Number` is encoded using variable-length encoding and `BigInt` is not. An `u32` fits into `f64`.

Reply:
If BigInto is not used to store integers (as we cast them to floats whenever possible), what's its purpose?

Reply:
Oh, didn't see this comment. It's just that `Any::Number` should be preferred. For i64, sure use `Any::BigInt` if you use more than 53 bits.